skimage/feature/corner.py

```diff
@@ -116,6 +119,14 @@ def structure_tensor(image, sigma=1, mode='constant', cval=0, order=None):
         else:
             order = 'rc'
 
+    if isinstance(sigma, tuple):
+        if len(sigma) <= 1:
+            raise TypeError('single sigma must be passed as float or int')
+        else:
+            if len(sigma) != image.ndim:
+                raise ValueError('sigma must have as many elements as image '
+                                 'has axes')
```

Comment:
I think that the special case of length 1 iterable is not that special and that we may allow passing array_like values and not only tuples
```suggestion
    if not np.isscalar(sigma):
        sigma = tuple(sigma)
        if len(sigma) != image.ndim:
            raise ValueError('sigma must have as many elements as image '
                             'has axes')
```

Reply:
Very true, thanks @rfezzani! I'll update the test.